### PR TITLE
fix: remove use of object ACLs

### PR DESCRIPTION
**Pull request by nosnilmot**
#### Description of changes

Best practice for restricting access to Amazon S3 content by using an origin access identity (OAI) is to us an S3 bucket policy, which is already the case.

Per-object ACLs are NOT recommended, see:

https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-restricting-access-to-s3.html#private-content-updating-s3-acls

#### Issue

This change is motivated by the fact that AWS is currently experiencing an issue in us-east-2 whereby CloudFront OAI S3CanonicalUserId are not being accepted in S3 PutObject requests, resulting in inability to publish amplify frontends to that region:

Publish started for S3AndCloudFront
✖ Error has occurred during file upload.
An error occurred during the publish operation: Invalid id
⚠️ Review the Amplify CLI troubleshooting guide for potential next steps: https://docs.amplify.aws/cli/project/troubleshooting/ 

(btw, that troubleshooting page was no help, and I had to modify amplify-cli source to identify what the source of the error was anyway)

#### Description of how you validated changes

`amplify-dev publish -c` works

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by danielleadams** [comment]
Hi @nosnilmot - thanks for your contribution! Running the test suite on it now.

**Comment by nosnilmot** [comment]
> Hi @nosnilmot - thanks for your contribution! Running the test suite on it now.

Hi @danielleadams - I see the CI tests have failed, but I do not think the failures are related to my PR. Please advise on how to proceed. Might rebasing help?

**Comment by danielleadams** [comment]
Hi @nosnilmot - if you don't mind rebasing, that would be helpful and I can rerun them. Thank you!

**Comment by nosnilmot** [comment]
@danielleadams I have rebased, and the automated CI tests have passed. Could you rerun the test suite please?

**Review by danielleadams — APPROVED** [review]

File: packages/amplify-category-hosting/lib/S3AndCloudFront/helpers/file-uploader.js [hunk 1]
```diff
@@ -75,12 +58,6 @@ async function uploadFile(s3Client, hostingBucketName, distributionDirPath, file
     ContentType: contentType || 'text/plain',
   };
 
-  if (cloudFrontS3CanonicalUserId) {
-    uploadParams.GrantRead = `id=${cloudFrontS3CanonicalUserId}`;
-  } else {
-    uploadParams.ACL = 'public-read';
```
Comment (danielleadams) [inline packages/amplify-category-hosting/lib/S3AndCloudFront/helpers/file-uploader.js:81]: Does the `uploadParams.ACL` need to be set still?

**Review by danielleadams — COMMENTED** [review]
@nosnilmot Approved, but I just had one question. Otherwise, changes look good.

**Review by danielleadams — APPROVED** [review]

**Comment by danielleadams** [comment]
@nosnilmot I'm running integration tests, but I would like to follow up with questions for other team members before we merge. I'll keep you posted on the progress of the PR.

**Comment by nosnilmot** [comment]
> @nosnilmot I'm running integration tests, but I would like to follow up with questions for other team members before we merge. I'll keep you posted on the progress of the PR.

@danielleadams any updates?

**Comment by nosnilmot** [comment]
@danielleadams ping?

**Comment by danielleadams** [comment]
@nosnilmot thanks for your patience here. I'm working through my backlog - so just getting around to this. Can you give me a week and let me see if I can move this forward? I will report back.

**Comment by danielleadams** [comment]
@nosnilmot thanks for your patience here - we're finally ready to merge this. I'm just running tests on the change now.

**Review by danielleadams — APPROVED** [review]

**Review by jhockett — APPROVED** [review]

File: packages/amplify-category-hosting/lib/S3AndCloudFront/helpers/file-uploader.js [hunk 2]
```diff
@@ -6,29 +6,24 @@ const sequential = require('promise-sequential');
...
 
-  let cloudFrontS3CanonicalUserId;
-  if (context.exeInfo.template.Resources.OriginAccessIdentity && output.CloudFrontOriginAccessIdentity) {
-    const cloudFrontClient = await getCloudFrontClient(context, 'retrieve');
-    const params = { Id: output.CloudFrontOriginAccessIdentity };
-    const originAccessIdentity = await cloudFrontClient.getCloudFrontOriginAccessIdentity(params).promise();
-    cloudFrontS3CanonicalUserId = originAccessIdentity.S3CanonicalUserId;
+  let hasCloudFront = false;
+  if (context.exeInfo.template.Resources.CloudFrontDistribution) {
+    hasCloudFront = true;
```
Comment (evcodes) [inline packages/amplify-category-hosting/lib/S3AndCloudFront/helpers/file-uploader.js:22]: let hasCloudFront = context.exeInfo.template.Resources.CloudFrontDistribution ? true : false;
```suggestion
let hasCloudFront = context.exeInfo.template.Resources.CloudFrontDistribution ? true : false;
```

Comment (nosnilmot) [reply packages/amplify-category-hosting/lib/S3AndCloudFront/helpers/file-uploader.js:22]: It's obviously a fine suggestion but if I commit it and reset the reviews/approvals will that set the PR back another 8 months?

**Comment by evcodes** [comment]
Hey @nosnilmot, your contribution was merged and released on 11.1.1. Thanks for your help 👍🏼

**Pull request closed without merging**